code/datums/mind.dm

```diff
@@ -110,22 +114,24 @@
 	memory = null
 
 // Datum antag mind procs
-/datum/mind/proc/add_antag_datum(datum_type, on_gain = TRUE)
+/datum/mind/proc/add_antag_datum(datum_type)
 	if(!datum_type)
 		return
-	if(!can_hold_antag_datum(datum_type))
-		return
 	var/datum/antagonist/A = new datum_type(src)
-	antag_datums += A
-	if(on_gain)
-		A.on_gain()
+	if(!A.can_be_owned(src))
+		qdel(A)
+		return
+	LAZYADD(antag_datums, A)
+	A.on_gain()

```

Comment:
you should never, NEVER not call on_gain; if you need to not do so, it should be done via a subtype with appropriate overrides, because on_gain is the only thing that currently handles antag huds.